tests/integ/test_query_plan_analysis.py

```diff
@@ -0,0 +1,543 @@
+    assert_df_subtree_query_complexity(df1, {PlanNodeCategory.COLUMN.value: 1})
+
+    split_to_table = table_function("split_to_table")
+
+    # SELECT "SEQ", "INDEX", "VALUE" FROM (
+    #  SELECT T_RIGHT."SEQ", T_RIGHT."INDEX", T_RIGHT."VALUE" FROM
+    #      (select 'James' as name, 'address1 address2 address3' as addresses) AS T_LEFT
+    #  JOIN  TABLE (split_to_table("ADDRESSES", ' ') ) AS T_RIGHT)
+    df2 = df1.select(split_to_table(col("addresses"), lit(" ")))
+    assert_df_subtree_query_complexity(
+        df2,
+        {
+            PlanNodeCategory.COLUMN.value: 8,
+            PlanNodeCategory.JOIN.value: 1,
+            PlanNodeCategory.FUNCTION.value: 1,
+            PlanNodeCategory.LITERAL.value: 1,
+        },
+    )
+
+    #  SELECT T_LEFT.*, T_RIGHT.* FROM (select 'James' as name, 'address1 address2 address3' as addresses) AS T_LEFT
+    #   JOIN  TABLE (split_to_table("ADDRESS", ' ')  OVER (PARTITION BY "LAST_NAME"  ORDER BY "FIRST_NAME" ASC NULLS FIRST)) AS T_RIGHT
+    df3 = df1.join_table_function(
+        split_to_table(col("address"), lit(" ")).over(
+            partition_by="last_name", order_by="first_name"
+        )
+    )
+    assert_df_subtree_query_complexity(
+        df3,
+        {
+            PlanNodeCategory.COLUMN.value: 6,
+            PlanNodeCategory.JOIN.value: 1,
+            PlanNodeCategory.FUNCTION.value: 1,
+            PlanNodeCategory.LITERAL.value: 1,
+            PlanNodeCategory.PARTITION_BY.value: 1,
+            PlanNodeCategory.ORDER_BY.value: 1,
+            PlanNodeCategory.WINDOW.value: 1,
+            PlanNodeCategory.OTHERS.value: 1,
+        },
+    )
+
+
+@pytest.mark.parametrize(
+    "set_operator", [SET_UNION, SET_UNION_ALL, SET_EXCEPT, SET_INTERSECT]
+)
+def test_set_operators(session: Session, sample_table: str, set_operator: str):
+    df1 = session.table(sample_table)
+    df2 = session.table(sample_table)
+    if set_operator == SET_UNION:
+        df = df1.union(df2)
+    elif set_operator == SET_UNION_ALL:
+        df = df1.union_all(df2)
+    elif set_operator == SET_EXCEPT:
+        df = df1.except_(df2)
+    else:
+        df = df1.intersect(df2)
+
+    # ( SELECT  *  FROM SNOWPARK_TEMP_TABLE_9DJO2Y35IT) set_operator ( SELECT  *  FROM SNOWPARK_TEMP_TABLE_9DJO2Y35IT)
+    assert_df_subtree_query_complexity(
+        df, {PlanNodeCategory.COLUMN.value: 2, PlanNodeCategory.SET_OPERATION.value: 1}
+    )
+
+
+def test_agg(session: Session, sample_table: str):
+    df = session.table(sample_table)
+    df1 = df.agg(avg("a"))
+    df2 = df.agg(avg("a") + 1)
+    df3 = df.agg(avg("a"), avg(col("b") + lit(1)).as_("avg_b"))
+    df4 = df.group_by(["a", "b"]).agg(avg("c"))
+
+    #  SELECT avg("A") AS "AVG(A)" FROM ( SELECT  *  FROM sample_table) LIMIT 1
+    assert_df_subtree_query_complexity(
+        df1,
+        {
+            PlanNodeCategory.COLUMN.value: 3,
+            PlanNodeCategory.LOW_IMPACT.value: 1,
+            PlanNodeCategory.FUNCTION.value: 1,
```

Comment:
oh, where does the function comes from? is that from aliasing function?

Reply:
it comes from AVG